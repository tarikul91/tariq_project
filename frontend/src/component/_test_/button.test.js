import { render, screen,cleanup } from "@testing-library/react";
import Button from "component/button";
import renderer from "react-test-renderer"
afterEach(cleanup)

test('Button test',()=>{
    render(<Button className='btn' >Save</Button>)
    expect(screen.getByTestId('btn-1')).toHaveTextContent('Save')
    expect(screen.getByTestId('btn-1')).toHaveClass('btn')
})
it('Button snapshot',()=>{
    const tree = renderer.create(<Button disabled>Save</Button>).toJSON()
    expect(tree).toMatchSnapshot()
})