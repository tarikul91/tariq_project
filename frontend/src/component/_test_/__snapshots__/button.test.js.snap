// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<button
  data-testid="btn-1"
  disabled={true}
>
  Save
</button>
`;
